libs/AddressProvider/src/main/java/finance/omm/libs/address/AddressProvider.java

```diff
@@ -65,5 +76,31 @@ protected void checkAddressProvider() {
         Context.require(Context.getCaller().equals(_addressProvider.get()), "require Address provider contract access");
     }
 
+    public void onlyOwner() {
+        Address sender = Context.getCaller();
+        Address owner = Context.getOwner();
+        if (!sender.equals(owner)) {
+            Context.revert(TAG + ": SenderNotScoreOwnerError:  (sender)"
+                    + sender + " (owner)" + owner);
+        }
+    }
+
+    public void onlyAddressProvider() {
+        Address addressProvider = getAddressProvider();
+        Address sender = Context.getCaller();
+        if (!sender.equals(addressProvider)) {
+            Context.revert(TAG + ": SenderNotAddressProviderError:  (sender)"
+                    + sender + " (address provider)" + addressProvider);
+        }
+    }
+
+    public void onlyGovernance() {
+        Address sender = Context.getCaller();
+        Address governance = getAddress("governance");
+        if (!sender.equals(governance)) {
+            Context.revert(TAG + ": SenderNotGovernanceError: (sender)" + sender
+                    + " (governance)" + governance);
+        }
+    }
```

Comment:
Nice to have these utility method in super class. However, We would like to have custom exception thrown like [here](https://github.com/openmoneymarket/openmoneymarket-contracts/blob/490e038ebff123859a513fc63bc1491b5832c7c3/core-contracts/RewardDistribution/src/main/java/finance/omm/score/core/reward/distribution/AbstractRewardDistribution.java#L400)
   ```
protected void checkGovernance() {
        if (!Context.getCaller()
                .equals(this.getAddress(Contracts.GOVERNANCE.getKey()))) {
            throw RewardDistributionException.notGovernanceContract();
        }
    }
```
So I think we can remove these method from here and implement in contract main class itself.

Reply:
what about passing the required Exception as argument? like

```
public void onlyGovernanceOrElseThrow(UserRevertException userRevertException) {
     if (!Context.getCaller()
             .equals(this.getAddress(Contracts.GOVERNANCE.getKey()))) {
         throw userRevertException;
     }
```
 }

Reply:
Yeah it seems to be  fine :)

Reply:
done!